benchmark/runbench.d

```diff
@@ -335,254 +45,154 @@ int main(string[] args)
-    string test_app_dmd_base = output_dir ~ envData.sep ~ baseName(test_name) /*~ "_"*/;
-
-    if (!std.file.exists(output_dir))
-        std.file.mkdirRecurse(output_dir);
-
-    TestArgs testArgs;
+    immutable bindir = absolutePath("bin");
 
-    gatherTestParameters(testArgs, input_dir, input_file, envData);
-
-    //prepare cpp extra sources
-    if (testArgs.cppSources.length)
+    foreach(ref src; sources)
     {
-        switch (envData.compiler)
-        {
-            case "dmd":
-                if(envData.os != "win32" && envData.os != "win64")
-                    testArgs.requiredArgs ~= " -L-lstdc++";
-                break;
-            case "ldc":
-                testArgs.requiredArgs ~= " -L-lstdc++";
-                break;
-            case "gdc":
-                testArgs.requiredArgs ~= "-Xlinker -lstdc++";
-                break;
-            default:
-                writeln("unknown compiler: "~envData.compiler);
-                return 1;
-        }
-        foreach (cur; testArgs.cppSources)
-        {
-            auto curSrc = input_dir ~ envData.sep ~"extra-files" ~ envData.sep ~ cur;
-            auto curObj = output_dir ~ envData.sep ~ cur ~ envData.obj;
-            string command = envData.ccompiler;
-            if (envData.compiler == "dmd")
-            {
-                if (envData.os == "win32")
-                {
-                    command ~= " -c "~curSrc~" -o"~curObj;
-                }
-                else if (envData.os == "win64")
-                {
-                    command ~= ` /c /nologo `~curSrc~` /Fo`~curObj;
-                }
-                else
-                {
-                    command ~= " -m"~envData.model~" -c "~curSrc~" -o "~curObj;
-                }
-            }
-            else
-            {
-                command ~= " -m"~envData.model~" -c "~curSrc~" -o "~curObj;
-            }
-
-            auto rc = system(command);
-            if(rc)
-            {
-                writeln("failed to execute '"~command~"'");
-                return 1;
-            }
-            testArgs.sources ~= curObj;
-        }
+        writeln("COMPILING ", src);
+        version (Windows) enum exe = "exe"; else enum exe = "";
+        auto bin = buildPath(bindir, src.chompPrefix("./").setExtension(exe));
+        auto cmd = std.string.format("%s %s -op -odobj -of%s %s", cfg.dmd, cfg.dflags, bin, src);
+        runCmd(cmd, cfg.verbose);
+        src = bin;
     }
-    writef(" . %-16s ", input_file);
-    fflush(core.stdc.stdio.stdout);
 
-    if (testArgs.disabled)
+    foreach(bin; sources)
     {
-        writefln("!!! [DISABLED: %s]", testArgs.disabled_reason);
-        return 0;
-    }
+        import std.datetime, std.algorithm : min;
+        auto sw = StopWatch(AutoStart.yes);
+        auto minDur = Duration.max;
 
-    removeIfExists(output_file);
+        stdout.writef("RUNNING %-20s", bin.relativePath(bindir));
```

Comment:
Can we please reduce the vertical space here for better alignment and more possible output by the GC profiling, e.g. 
```
stdout.writef("R %-16s", bin.baseName.stripExtension);
```